test/log.ts

```diff
@@ -338,16 +343,48 @@ describe('Log', () => {
   });
 
   describe('write', () => {
-    const ENTRY = {} as Entry;
-    const OPTIONS = {};
+    let ENTRY: Entry;
+    let ENTRIES: Entry[];
+    let OPTIONS: WriteOptions;
     const FAKE_RESOURCE = 'fake-resource';
+    let truncateEntriesStub: sinon.SinonStub;
+    let decorateEntriesStub: sinon.SinonStub;
+    let fakeResourceStub: sinon.SinonStub;
+    let projectIdStub: sinon.SinonStub;
```

Comment:
I don't love that this adds a lot of logic related to `log` both in the scope of the `describe`, and in the `beforeEach`, I would be tempted to pull this logic into a helper that's something like:

```
function stubLogger(log: Log) {
  // all the mocking logic here.
}
function restoreLogger(log: Log) {

}
```

and I would keep this logic close to:

```js
  before(() => {
    Log = proxyquire('../src/log', {
      '@google-cloud/promisify': fakeCallbackify,
      './entry': {Entry},
      './metadata': fakeMetadata,
    }).Log;
    const assignSeverityToEntries_ = Log.assignSeverityToEntries_;
    Log.assignSeverityToEntries_ = (...args) =>
      (assignSeverityToEntriesOverride || assignSeverityToEntries_).apply(
        null,
        args
      );
  });
```

Sot that it's clear what the heck `log` is, it's a heavily modified Logger instance.

Reply:
I was trying to enhance the test cases for `write` without refactoring the entire `Log` suite, oh well!

I went ahead and moved the common logger stubs to the root suite. Some of them need to be maintained within their respective sub-suites, e.g. `write`'s stub for `truncateEntries`. I also changed the rest of the tests based on the revised stubbing.